addons/wowl/static/src/components/dropdown/dropdown_item.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from "@odoo/owl";
+
+export enum ParentClosingMode {
+  None = "none",
+  ClosestParent = "closest",
+  AllParents = "all",
+}
+
+export interface DropdownItemClickedPayload {
+  payload: Object;
+  dropdownClosingRequest: {
+    mode: ParentClosingMode;
+    isFresh: boolean;
+  };
+}
+
+export class DropdownItem extends Component {
+  static template = "wowl.DropdownItem";
+  static props = {
+    onClickPayload: {
+      type: Object,
+      optional: true,
+    },
+    parentClosingMode: {
+      type: ParentClosingMode,
+      optional: true,
+    },
+  };
+  static defaultProps = {
+    parentClosingMode: ParentClosingMode.AllParents,
+  };
+
+  /**
+   * Handlers
+   */
+  fire(ev: MouseEvent) {
+    const payload: DropdownItemClickedPayload = {
+      payload: this.props.onClickPayload,
+      dropdownClosingRequest: {
+        isFresh: true,
+        mode: this.props.parentClosingMode,
+      },
+    };
+    this.trigger("dropdown-item-clicked", payload);
+
+    // Stopping propagation as we replaced the event with a custom one
+    ev.stopPropagation();
```

Comment:
I thought the idea of triggering a new event was partially avoid having to stop the propagation of a Dom Event ? You sure you need to stop it ?

Reply:
This is not needed, you are correct :-)